lib/matplotlib/legend.py

```diff
@@ -743,6 +743,7 @@ def _init_legend_box(self, handles, labels, markerfirst=True):
         label_prop = dict(verticalalignment='baseline',
                           horizontalalignment='left',
                           fontproperties=self.prop,
+                          axes=self.axes
```

Comment:
I guess that is one soon.  However, why doesn't it get set when the artist is added to the axes?

Reply:
I've updated it to set the `axes` of `._legend_box`, which automatically sets the `axes` of its children. I think this is more general solution?

Reply:
That _sounds_ better!